### PR TITLE
upload hw 9

### PR DESCRIPTION
**Pull request by proevgenii**

### CODE REVIEW

File: hw9/hw1.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+4
+6
+
+>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
+[1, 2, 3, 4, 5, 6]
+"""
+from contextlib import ExitStack
+from itertools import cycle, islice
+from pathlib import Path
+from typing import Iterator, List, Union
```
Comment (PeppaTheC) [inline hw9/hw1.py]: is list union used?

File: hw9/hw1.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            for next in nexts:
+                yield int(next().strip("\n"))
+        except StopIteration:
+            num_active -= 1
+            nexts = cycle(islice(nexts, num_active))
+
+
+def opener(file_list: [Path, str]) -> Iterator:
+    with ExitStack() as stack:
+        files = [stack.enter_context(open(name)) for name in file_list]
```
Comment (PeppaTheC) [inline hw9/hw1.py:37]: when file is closed?

File: hw9/hw1.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    while num_active:
+        try:
+            for next in nexts:
+                yield int(next().strip("\n"))
+        except StopIteration:
+            num_active -= 1
+            nexts = cycle(islice(nexts, num_active))
+
+
+def opener(file_list: [Path, str]) -> Iterator:
```
Comment (PeppaTheC) [inline hw9/hw1.py:35]: is it used?

File: hw9/hw1.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+1
+3
+5
+
+file2.txt:
+2
+4
+6
+
+>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
```
Comment (PeppaTheC) [inline hw9/hw1.py:14]: does this example works?

File: hw9/hw2.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        self.er_name = er_name
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return exc_type is self.er_name
+
+
+with Suppressor(IndexError):
```
Comment (PeppaTheC) [inline hw9/hw2.py]:
```
class MyIndexError(IndexError):
    pass

with Suppressor(IndexError):
    raise MyIndexErro
```
must work as well

File: hw9/hw3.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                    result += len(list(fi))
+
+                else:
+                    text = fi.read()
+                    result = list(map(tokenizer, text))[-1][0]
+
+    return result
+
+
+def universal_file_counter(
```
Comment (PeppaTheC) [inline hw9/hw3.py]: remove

File: hw9/hw3.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+from pathlib import Path
+from typing import Callable, Optional
+
+
+def universal_file_counter(
+    dir_path: Path, file_extension: str, tokenizer: Optional[Callable] = None
+) -> int:
+    result = 0
+    files_list = []
+    for file in os.listdir(dir_path):
```
Comment (PeppaTheC) [inline hw9/hw3.py]: use rglob to simplify

Comment (PeppaTheC) [reply hw9/hw3.py]: from pathlib

**Review by PeppaTheC — CHANGES_REQUESTED** [review]

File: hw9/hw1.py (hunk 3 above)
Comment (proevgenii) [reply hw9/hw1.py:35]: what exactly? opener function ?

File: hw9/hw1.py (hunk 4 above)
Comment (proevgenii) [reply hw9/hw1.py:14]: yes 
It runs as :
print(list(opener(["file1.txt", "file2.txt"])))

File: hw9/hw2.py (hunk 5 above)
Comment (proevgenii) [reply hw9/hw2.py]: fixed

File: hw9/hw1.py (hunk 1 above)
Comment (proevgenii) [reply hw9/hw1.py]: fixed

File: hw9/hw1.py (hunk 2 above)
Comment (proevgenii) [reply hw9/hw1.py:37]: All opened files will automatically be closed at the end of the with statement, even if attempts to open files later in the list raise an exception

File: hw9/hw3.py (hunk 6 above)
Comment (proevgenii) [reply hw9/hw3.py]: fixed

**Review by PeppaTheC — APPROVED** [review]

**Pull request merged**